apps/emqx_plugins/src/emqx_plugins.erl

```diff
@@ -395,15 +395,18 @@ get_config_bin(NameVsn) ->
 %% RPC call from Management API or CLI.
 %% The plugin config Json Map was valid by avro schema
 %% Or: if no and plugin config ALWAYS be valid before calling this function.
-put_config(NameVsn, ConfigJsonMap, AvroValue) when (not is_binary(NameVsn)) ->
-    put_config(bin(NameVsn), ConfigJsonMap, AvroValue);
-put_config(NameVsn, ConfigJsonMap, _AvroValue) ->
+put_config(NameVsn, ConfigJsonMap) when (not is_binary(NameVsn)) ->
+    put_config(bin(NameVsn), ConfigJsonMap);
+put_config(NameVsn, ConfigJsonMap) ->
     HoconBin = hocon_pp:do(ConfigJsonMap, #{}),
-    ok = backup_and_write_hocon_bin(NameVsn, HoconBin),
     %% TODO: callback in plugin's on_config_upgraded (config vsn upgrade v1 -> v2)
-    ok = maybe_call_on_config_changed(NameVsn, ConfigJsonMap),
-    ok = persistent_term:put(?PLUGIN_PERSIS_CONFIG_KEY(NameVsn), ConfigJsonMap),
-    ok.
+    case maybe_call_on_config_changed(NameVsn, ConfigJsonMap) of
```

Comment:
I am not sure this will work, because put_config is used to make initial configuration _before_ the app is loaded and started, so the call will always fail/

Reply:
Also, there are no tests in the base branch to verify the change 😓